connectivity/tests/interruption.go

```diff
@@ -87,17 +137,28 @@ func (n *noInterruptedConnections) Run(ctx context.Context, t *check.Test) {
 	if err != nil {
 		t.Fatalf("Failed to read interruption test result files: %s", err)
 	}
-	prevRestartCount := make(map[string]string)
-	if err := gojson.Unmarshal(b, &prevRestartCount); err != nil {
+	prevCounts := interruptionCounts{}
+	if err := gojson.Unmarshal(b, &prevCounts); err != nil {
 		t.Fatalf("Failed to unmarshal JSON test result file: %s", err)
 	}
 
-	for pod, count := range restartCount {
-		if prevCount, found := prevRestartCount[pod]; !found {
+	for pod, restarts := range counts.Restarts {
+		if prevRestarts, found := prevCounts.Restarts[pod]; !found {
 			t.Fatalf("Could not found Pod %s restart count", pod)
-		} else if prevCount != count {
+		} else if prevRestarts != restarts {
 			t.Fatalf("Pod %s flow was interrupted (restart count does not match %s != %s)",
-				pod, prevCount, count)
+				pod, prevRestarts, restarts)
+		}
+	}
+
+	if ct.Params().InterruptionTestForIPsecKeyRotation {
+		for node, xfrmErrors := range counts.XfrmErrors {
+			if prevXfrmErrors, found := prevCounts.XfrmErrors[node]; !found {
+				t.Fatalf("Could not found Node %s xfrm errors", node)
+			} else if prevXfrmErrors != xfrmErrors {
```

Comment:
I'm unsure we have a guarantee that we will always dump the XFRM errors in the same order. I believe we just print them in the order the the kernel shows them.

Reply:
I sorted them in ` collectXfrmErrors()`:

```go
		sort.Strings(xErrors)
		xfrmErrors[node.GetName()] = strings.Join(xErrors, "\n")
```

Reply:
I totally missed that :facepalm: